docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,200 @@
+1. Process that information (and optionally respond), possibly connecting to other services within or outside the solution, using appropriate credentials, to update state related to that request.
+
+There are numerous services you can use with that basic pattern – HTTP requests, queues and message busses, event stores, various flavours of databases and storage, caching. The details of these services is outside the scope of this discussion, but they are all consumed using this basic pattern.
+
+In a multi-tenanted solution, you need to layer a notion of "the intended tenant" into your design:
+
+1. Receive a stimulus of some kind, and dispatch it to the appropriate tenant
+1. Connect to and authenticate against one or more services within or outside the tenanted solution using credentials appropriate for the given tenant, in order to obtain the information required to respond to the stimulus.
+1. Process that information (and optionally respond) using code and configuration appropriate for that tenant, possibly connecting to the correct instance of other services within or outside the tenanted solution to update state related to that request, using the instances and credentials appropriate for that tenant.
+
+The way in which this is done is very technology specific, but in essence the environment hosting the tenancy needs to:
+
+1. Identify the intended tenant for the stimulus
+1. Identify the code that is needed to handle the stimulus, and the compute resource to which that stimulus should be directed
+1. Provide a means for that code to find and connect to the services it uses.
+
+There are common patterns that can help a multi-tenanted solution perform these tasks, and abstract as much of this from the code executing in the tenant as possible.
+
+Typically this will include
+1. code at the stimulus/event routing layer (e.g. http routing, message handlers)
+1. factories to be used by the client code to create tenant-specific instances of service clients, configured appropriately
+
+## Solution-layer concerns
+As we dig down in the stack, we finally reach implementation concerns in the multi-tenanted solution itself.
+
+How do you design a SQL Azure database to host multi-tenanted data? What are the isolation characteristics? What about Blob storage? Cosmos DB? 
+
+These solution-level design choices are well explored in the Azure documentation for each individual service.
+
+For example
+- [Azure SQL Database]( https://docs.microsoft.com/en-us/azure/azure-sql/database/saas-tenancy-app-design-patterns)
+- [Cosmos DB]( https://docs.microsoft.com/en-us/azure/architecture/guide/multitenant/service/cosmos-db)
+- [Azure Storage]( https://docs.microsoft.com/en-us/azure/architecture/guide/multitenant/service/storage)
+
+There is also some [high-level conceptual documentation](https://docs.microsoft.com/en-gb/azure/architecture/guide/multitenant/considerations/overview) about multi-tenanted solution architecture.
```

Comment:
This came out on Friday, might be worth linking to as well https://www.youtube.com/watch?v=aem8elgN7iI